boomer/src/main.rs

```diff
@@ -36,17 +36,24 @@ struct CliArgs {
     )]
     device: PathBuf,
 
+    #[arg(short = 'C', long, help = "Connector name, for example: HDMI-A-1")]
+    connector: Option<String>,
+
     #[arg(short, long, action = clap::ArgAction::Count)]
     verbose: u8,
 }
 
-fn find_connector(device: &Device) -> Option<Rc<Connector>> {
+fn find_connector(device: &Device, connector_name: &Option<String>) -> Option<Rc<Connector>> {
     device
         .connectors()
         .filter(|con| match con.connector_type() {
             ConnectorType::HDMIA | ConnectorType::HDMIB => true,
             _ => false,
         })
+        .filter(|con| match connector_name {
+            Some(name) => name == &con.to_string(),
+            None => true,
+        })
```

Comment:
I don't think we need to chain the filters/find. Also, if the connector name is passed, we shouldn't try anything else, we should just try this name, and fail if it doesn't exist.

So something like 

```rust
    if let Some(name) = connector_name {
        device
            .connectors()
            .find(|con| con.to_string().as_str() == name)
    } else {
        device.connectors().find(|con| {
            con.connector_type() == ConnectorType::HDMIA
                && con.status().unwrap_or(ConnectorStatus::Unknown) == ConnectorStatus::Connected
        })
    }

```